lib/auth/apikey.ex

```diff
@@ -9,57 +12,79 @@ defmodule Auth.Apikey do
   schema "apikeys" do
     field :client_secret, :binary
     field :client_id, :binary
-    field :description, :string
-    field :name, :string
-    field :url, :binary
     field :person_id, :id
     field :status, :id
+    belongs_to :app, Auth.App
 
     timestamps()
   end
 
-  @doc false
-  def changeset(apikey, attrs) do
-    apikey
-    |> cast(attrs, [:client_id, :client_secret, :name, :description, :url, :person_id])
-    |> validate_required([:client_secret])
+  @doc """
+  `encrypt_encode/1` does exactly what it's name suggests,
+  AES Encrypts a string of plaintext and then Base58 encodes it.
+  We encode it using Base58 so it's human-friendly (readable).
+  """
+  def encrypt_encode(plaintext) do
+    plaintext |> Fields.AES.encrypt() |> Base58.encode()
   end
 
-  def change_apikey(%Apikey{} = apikey) do
-    Apikey.changeset(apikey, %{})
+  @doc """
+  `create_api_key/1` uses the `encrypt_encode/1` to create an API Key
+  that is just two strings joined with a forwardslash ("/").
+  This allows us to use a *single* environment variable.
+  """
+  def create_api_key(id) do
+    encrypt_encode(id) <> "/" <> encrypt_encode(id)
   end
 
-  def create_apikey(attrs \\ %{}) do
-    %Apikey{}
-    |> Apikey.changeset(attrs)
-    |> Repo.insert()
+  @doc """
+  `decode_decrypt/1` accepts a `key` and attempts to Base58.decode
+  followed by AES.decrypt it. If decode or decrypt fails, return 0 (zero).
+  """
+  def decode_decrypt(key) do
+    try do
+      key |> Base58.decode() |> Fields.AES.decrypt() |> String.to_integer()
+    rescue
+      ArgumentError ->
+        0
```

Comment:
Any reason for returning `0` and not `{:error, :reason}`?

Reply:
Yeah, I honestly don't mind refactoring this. I just thought the `0` failure was OK for MVP. 
Again, this is a good area for improvement as is adding Logging for monitoring for the errors. 👍